llvm/test/Transforms/LoopIdiom/basic.ll

```diff
@@ -596,7 +594,7 @@ for.end:                                          ; preds = %for.body
 define void @test13_pattern(ptr nocapture %P) nounwind ssp {
 ; CHECK-LABEL: @test13_pattern(
 ; CHECK-NEXT:  entry:
-; CHECK-NEXT:    call void @memset_pattern16(ptr [[P:%.*]], ptr @.memset_pattern.1, i64 80000)
+; CHECK-NEXT:    call void @llvm.experimental.memset.pattern.p0.p0.i64(ptr align 4 [[P:%.*]], ptr @G, i64 10000, i1 false)
```

Comment:
Just to note, you dropped the pointer to int conversion from the prior version of the patch, and thus until a reworked https://github.com/llvm/llvm-project/pull/138559 lands we have an aliasing imprecision here.  I'm explicitly okay with that being true for the landed patch, and will follow up on my patch post-commit.